apluslms_roman/builder.py

```diff
@@ -10,18 +11,33 @@
 
 class Builder:
     def __init__(self, engine, config, observer=None):
-        assert isdir(config.__path__), "Course configuration path doesn't exists"
+        # if not isdir(config.__path__):
+        if not isdir(config.dir):
+            raise ValueError("config.dir isn't a directory.")
         self.config = config
-        self.path = config.__path__
+        self.path = config.dir
         self._engine = engine
         self._observer = observer or StreamObserver()
 
-    def build(self):
+
+    def get_steps(self, refs: list = None):
+        steps = [BuildStep.from_config(i, step) for i, step in enumerate(self.config.steps)]
+        if refs:
+            name_dict = {step.name: step for step in steps}
+            step_list = [int(ref) if ref.isdigit() else ref for ref in refs]
+            try:
+                steps = [steps[ref] if isinstance(ref, int) else name_dict[ref] for ref in step_list]
+            except KeyError as e:
+                raise ValueError("No step named " + str(e))
```

Comment:
These strings should be wrapped with `_(...)`, so when translations are introduced, they are automatically included. So, `_("No step named '{}'").format(e)`.